Pipes - Amy Cash - Solar System.rb

```diff
@@ -0,0 +1,196 @@
+#Wave Three (Wave One Hash Version follows, but is commented out; Wave Two is encapsulated in Wave Three)
+
+class Planet
+
+  attr_reader :name, :namesake, :type, :year, :distance 
+
+  def initialize(name, namesake, type, year, distance)
+    @name = name
+    @namesake = namesake
+    @type = type
+    @year = year
+    @distance = distance
+  end
+
+  def give_info
+    list =  "#{@name}: \tNamed for: #{@namesake} \tType: #{type} \tYear: #{year} Earth Years \tDistance: \t#{distance} million kilometers."
+    return list
+  end
+
+end
+
+class SolarSystem
+
+  def initialize(list_of_planets)
+    @planets = list_of_planets
+  end
+
+  def add_planet(planet)
+    @planets += planet
+  end
+
+  def print_list
+    list = ""
+    @planets.each_with_index do |planet, index|
+      list += @planets[index].give_info + "\n"
+    end
+    return list
+  end
+
+  def check_input(input)
+    until input.is_a?(Integer) && input.between?(1, 9)
+      print "Please enter a number between 1 and 9: "
+      input = gets.chomp.to_i
+    end
+  end
+
+  def select_planet
+    puts "Which Planet do you want to learn about? Choose from the following options: 1. Mercury, 2. Venus, 3. Earth, 4. Mars, 5. Jupiter, 6. Saturn, 7. Uranus, 8. Neptune, 9. Pluto "
+    answer = gets.chomp.to_i
+
+    check_input(answer)
+
+    choice = @planets[answer - 1]
+
+    puts "You chose #{choice.name}. Here's what I know about #{choice.name}: \n\n"
+
+    list = "#{choice.name} was named for #{choice.namesake}. Its year is #{choice.year} earth years long. It is #{choice.distance} million kilometers from the sun and it is #{choice.type}.\n\n"
+
+    return list
+  end
+
+  def find_distance_between_planets
+    puts "Now you can tell the distance between two planets: "
+
+    puts "\nWhat is the first planet you want to compare. Choose from the following options: 1. Mercury, 2. Venus, 3. Earth, 4. Mars, 5. Jupiter, 6. Saturn, 7. Uranus, 8. Neptune, 9. Pluto "
+    planet_1 = gets.chomp.to_i
+
+    check_input(planet_1)
+
+    puts "\nWhat is the second planet you want to compare. Choose from the following options: 1. Mercury, 2. Venus, 3. Earth, 4. Mars, 5. Jupiter, 6. Saturn, 7. Uranus, 8. Neptune, 9. Pluto "
+    planet_2 = gets.chomp.to_i
+
+    check_input(planet_2)
```

Comment:
You've almost exactly the same code here twice - that's usually a good indicator that extracting this functionality to a method might be appropriate.